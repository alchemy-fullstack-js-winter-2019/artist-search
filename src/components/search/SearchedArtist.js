import React from 'react';
import PropTypes from 'prop-types';
import { Link, BrowserRouter as Router } from 'react-router-dom';


function SearchedArtist({ values }) {
    const { id, name } = values;

    
    return (
        <>
        <Router>
            <Link to={`/${name}/${id}`}>
                <h2>{name}</h2>
                <h2>{id}</h2>
            </Link>
        </Router>
        </>
    );
}
SearchedArtist.propTypes = {
    values: PropTypes.object,
};

export default SearchedArtist;

