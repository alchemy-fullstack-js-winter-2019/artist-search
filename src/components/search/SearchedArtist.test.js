import React from 'react';
import renderer from 'react-test-renderer';
import SearchedArtist from './SearchedArtist';

jest.mock('../services/musicbrainApi.js');

describe('SearchedArtist', () => {
    it('matches a snapshot', () => {
        const values = {
            name: 'Nirvana',
            id: '5b11f4ce-a62d-471e-81fc-a69a8278c7da'
        };
        const tree = renderer.create(
            <SearchedArtist values={values} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
