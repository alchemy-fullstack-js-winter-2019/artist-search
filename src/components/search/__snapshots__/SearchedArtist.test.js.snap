// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchedArtist matches a snapshot 1`] = `
<a
  href="/Nirvana/5b11f4ce-a62d-471e-81fc-a69a8278c7da"
  onClick={[Function]}
>
  <h2>
    Nirvana
  </h2>
  <h2>
    5b11f4ce-a62d-471e-81fc-a69a8278c7da
  </h2>
</a>
`;
